metrics-healthchecks/src/main/java/com/codahale/metrics/health/HealthCheck.java

```diff
@@ -146,11 +185,83 @@ public String toString() {
             if (error != null) {
                 builder.append(", error=").append(error);
             }
+            if (details != null) {
+                Iterator<Map.Entry<String, Object>> it = details.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry<String, Object> e = it.next();
+                    builder.append(e.getKey())
+                        .append("=")
+                        .append(e.getValue().toString());
+                }
+            }
             builder.append('}');
             return builder.toString();
         }
     }
 
+	/**
+     * This a convenient builder for an {@link HealthCheck.Result}. It can be health (with optional message and detail)
+     * or unhealthy (with optional message, error and detail)
+     */
+    public static class ResultBuilder {
+        private boolean healthy;
+        private String message;
+        private Throwable error;
+        private Map<String, Object> details;
+
+        protected ResultBuilder(boolean healthy) {
+            this.details = new LinkedHashMap<String, Object>();
```

Comment:
The problem I have with it is that it allocates memory that we really don't need to allocate. Yes, it's a small amount of memory (my math says ~104 bytes), but it's also very easy to avoid doing it.